src/Validator/Rules/ValuesOfCorrectType.php

```diff
@@ -77,6 +78,8 @@ public function getVisitor(ValidationContext $context)
 
                     return Visitor::skipNode();
                 }
+
+                return null;
```

Comment:
I'd invert the condition so the `skipNode()` is in the main path and `null` as early exit.

Reply:
Will address in future PR.